File: Data Structures & Algorithms/remove-element/submission-0.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        new = []
        count = 0
        for num in nums:
            if num != val:
                new.append(num)
                count += 1

        for i in range(0, count):
            nums[i] = new[i]

        # nums = new # doesn't work

        return count

File: Data Structures & Algorithms/counting-bits/submission-0.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        std::vector<int> v;
        int count;

        for (int curr = 0; curr <= n; curr++) {
            count = 0;

            for (int i = 0; i < 32; i++) {
                if (curr & (1 << i)) {
                    count += 1;
                }
            }

            v.push_back(count);
        }

        return v;
    }
};

// 4 --> 100
// 5 --> 101
// 6 --> 110
// 7 --> 111